example/ingest.py

```diff
@@ -235,22 +316,24 @@ def generate_batches(it, chunk_size):
         yield cur_batch
 
 
+def _calculate_log_level(the_level) -> int:
+    if isinstance(the_level, str):
+        _level_str = str(the_level).lower()
+        if 'debug' in _level_str:
+            return DEBUG
+    return DEFAULT_LOG_LEVEL
+
+
 if __name__ == '__main__':
-    with ProcessPoolExecutor(max_workers=8) as pool:
-        print("Indexing BNL/L'Union articles")
-        futs = []
-        bnl_iter = bnl_load_documents(Path(LUNION_PATH))
-        for idx, batch in enumerate(generate_batches(bnl_iter, 1000)):
-            futs.append(pool.submit(index_documents, batch))
-            print("\r{:05}/{}".format((idx+1)*1000, LUNION_NUM_ARTICLES), end='')
-        for fut in as_completed(futs):
-            fut.result()
-        print("\nIndexing Google 1000 Books volumes")
-        futs = []
-        gbooks_iter = gbooks_load_documents(Path(GOOGLE1000_PATH))
-        for idx, batch in enumerate(generate_batches(gbooks_iter, 4)):
-            futs.append(pool.submit(index_documents, batch))
-            print("\r{:04}/{}".format((idx+1)*4, GOOGLE1000_NUM_VOLUMES), end='')
-        for fut in as_completed(futs):
-            fut.result()
-    print("\n")
+    PARSER = ArgumentParser(description='ingest example data into SOLR')
+    PARSER.add_argument('--log-level', help='like "debug", "info", "error" (default:info)', required=False,
+                        default=DEFAULT_LOG_LEVEL)
+    PARSER.add_argument('--num-workers', help="how many concurrent processes to start (default:4)", required=False,
+                        default=DEFAULT_N_WORKERS, type=int)
+    PARSER.add_argument('--books-only', help="if only interested in book corpus (default: False)", required=False, action='store_true')
+    ARGS = PARSER.parse_args()
+    LOGGER = Logger(LOGGER_NAME, _calculate_log_level(ARGS.log_level))
+    STDOUT_HANDLER = StreamHandler(sys.stdout)
+    STDOUT_HANDLER.setFormatter(Formatter("%(asctime)s [%(levelname)s] %(message)s", datefmt="%Y-%m-%dT%H:%M:%S"))
```

Comment:
I'd probably stick to `logging.basicConfig` and not even create a dedicated logger, since this is just a small utility script:

```suggestion
   logging.basicConfig(level=logging.DEBUG if args.debug else logging.WARN)
```

then you can just call `logging.debug` in your code.